test/parse/expr/bad-callref-int32.txt

```diff
@@ -0,0 +1,15 @@
+;;; TOOL: wat2wasm
+;;; ARGS: --enable-function-references
+;;; ERROR: 1
+(module
+  (func (export "main") (result i32)
+    (call_ref (i32.const 10)
+              (i32.const 13)
+    )
+  )
+)
+(;; STDERR ;;;
+out/test/parse/expr/bad-callref-int32.txt:6:6: error: type mismatch in call_ref, expected reference but got [i32]
```

Comment:
It looks like both of these parse failure tests (this one and `bad-callref-null.txt`) are checking type errors, so maybe they should go in `wabt/test/typecheck/` instead?

Reply:
sure, I've moved them